cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java

```diff
@@ -212,6 +214,19 @@ public void testDataset() throws Exception {
     DatasetTypeMeta datasetType = datasetTypeClient.list().get(0);
     testCommandOutputContains(cli, "create dataset instance " + datasetType.getName() + " " + datasetName,
                               "Successfully created dataset");
+    testCommandOutputContains(cli, "list dataset instances", FakeDataset.class.getSimpleName());
+
+    NamespaceClient namespaceClient = new NamespaceClient(cliConfig.getClientConfig());
+    namespaceClient.create(new NamespaceMeta.Builder().setId("bar").build());
+    cliConfig.setCurrentNamespace("bar");
+    // list of dataset instances is different in 'foo' namespace
+    testCommandOutputNotContains(cli, "list dataset instances", FakeDataset.class.getSimpleName());
+
+    // also can not create dataset instances if the type it depends on exists only in a different namespace.
+    testCommandOutputContains(cli, "create dataset instance " + datasetType.getName() + " " + datasetName,
```

Comment:
That scenario is already tested in cdap-clients test. And since cli uses that, I think it'd be redundant to test it again here.